app/src/main/java/com/fivegmag/a5gmsmediasessionhandler/MediaSessionHandlerMessengerService.kt

```diff
@@ -123,6 +139,15 @@ class MediaSessionHandlerMessengerService() : Service() {
                     val resource =
                         handleServiceAccessResponse(response, sendingUid, provisioningSessionId)
 
+                    // initialize Retrofit for ConsumptionReporting
+                    currentServiceAccessInformation = resource
```

Comment:
> handleServiceAccessResponse

Because in some Consumption reporting related methods such as  IsConsumptionReportingActivated and NeedReportConsumption, clientConsumptionReportingConfiguration of ServiceAccessInformation is need. So I need to save it as a member of the class MediaSessionHandlerMessengerService. I can only use resource directly only in handleStartPlaybackByServiceListEntryMessage.

Reply:
> We are maintaining multiple concurrent sessions in the Media Session Handler. This means that multiple applications can be connected to the same Media Session Handler (see handling in `handleServiceAccessResponse`). There is no need to define a variable `currentServiceAccessInformation`, you can use `resource`directly.

Because in some Consumption reporting related methods such as IsConsumptionReportingActivated and NeedReportConsumption, clientConsumptionReportingConfiguration of ServiceAccessInformation is need. So I need to save it as a member of the class MediaSessionHandlerMessengerService. I can only use resource directly only in handleStartPlaybackByServiceListEntryMessage.

Reply:
That is what `clientsSessionData` can be used for:

`clientsSessionData[sendingUid]?.serviceAccessInformation`

Reply:
Got it, thanks.  Done